sda-commons-server-openapi/src/main/java/org/sdase/commons/server/openapi/OpenApiBundle.java

```diff
@@ -133,6 +124,50 @@ public void run(Configuration configuration, Environment environment) {
         resourcePackages);
   }
 
+  private ContextIdOpenApiResource createOpenApiResource(String instanceId) {
+    var openApiConfiguration =
+        new SwaggerConfiguration()
+            .openAPI(existingOpenAPI != null ? existingOpenAPI : new OpenAPI())
+            .prettyPrint(true)
+            .resourcePackages(resourcePackages)
+            .readAllResources(false)
+            .sortOutput(true)
+            .filterClass(OpenAPISpecFilterSet.class.getName());
+
+    ContextIdOpenApiResource result = new ContextIdOpenApiResource();
+    result.contextId(OPENAPI_CONTEXT_ID_PREFIX + instanceId);
+    result.openApiConfiguration(openApiConfiguration);
+    return result;
+  }
+
+  /**
+   * This method can be used to generate the OpenAPI specification as JSON or YAML.
+   *
+   * @param type the type of the specification, either 'json' or 'yaml'
+   * @return the OpenAPI specification as JSON or YAML
+   */
+  public String generateOpenApi(String type) throws Exception {
```

Comment:
I would prefer two dedicated methods `generateOpenApiAsYaml()` and `generateOpenApiAsJson()` (or only yaml) so I don't have to mention, that the parameter should be an enum.

Reply:
Sure, no problem. It's less guessing and effort for consumers if they conveniently choose between the two. 👍